Integrate clearml

skinan opened the pull request:
Check logging works or not.

skinan committed:
ClearML Integration (with Tensorboard)

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,52 @@
+
+from __future__ import annotations
+
+from .tensorboard_handlers import TensorBoardHandler, TensorBoardImageHandler, TensorBoardStatsHandler
```
You're not using `TensorBoardHandler` right?

skinan replied:
No, I am not using Tensorboard Handler currently, I would remove it from here then.

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,52 @@
+    Class to write tensorboard stats into ClearML WebUI.
+    """
+
+    def __init__(self, project_name ="Muhammad/monai", task_name="unet_training_dict", output_uri=True, *args, **kwargs):
```
Please change the default project name :) Maybe something like `Monai AI`.
Also give it a general task name.

skinan replied:
Yeah, the default project name will be changed. The current is one given for draft pull-request. As there is permission issues, in our allegro master.

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,52 @@
+    Class to write tensorboard stats into ClearML WebUI.
+    """
+
+    def __init__(self, project_name ="Muhammad/monai", task_name="unet_training_dict", output_uri=True, *args, **kwargs):
```
Please also include additional arguments like tags for example. Just go though the Task.init arguments in the docs and see what could be useful here.

skinan replied:
Got it!

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,52 @@
+    Class to write tensorboard image stats into ClearML WebUI.
+    """
+
+    def __init__(self, project_name ="Muhammad/monai", task_name="unet_training_dict", output_uri=True, *args, **kwargs):
```
Same here :) No default Muhammad

skinan replied:
Will update it at the end before the final pull-request.

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
Updates on ClearML Integration

skinan committed:
default_tag

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -10,43 +10,143 @@
+                                    The subfolder structure is the following: <output destination name> / <project name> / <task name>.<Task ID>. Default set to 'True'
+        tags:                       Add a list of tags (str) to the created Task. For example: tags=['512x512', 'yolov3']. Default set to empty Python List.
+        reuse_last_task_id:         Force a new Task (experiment) with a previously used Task ID, and the same project and Task name. Default set to 'True'.
+        conitnue_last_task:         Continue the execution of a previously executed Task (experiment). Default set to 'False'.
```
typo: `continue`

skinan replied:
Thanks!

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -10,43 +10,143 @@
             from clearml import Task
         except ModuleNotFoundError:
-            raise "Please install ClearML with 'pip install clearml' before using it."
+            raise ModuleNotFoundError(f"Please install ClearML with 'pip install clearml' before using it.")
```
This does not need to be an f-string

skinan replied:
Okay!

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,152 @@
+            self.clearml_task = Task.current_task()
+        else:
+            self.clearml_task = Task.init(
+                                            project_name=project_name, 
```
This is weird indentation, please run a linter on the code to make sure it is formatted nicely

thepycoder commented on monai/handlers/clearml_handlers.py:
```diff
@@ -0,0 +1,152 @@
+    """
+
+    def __init__(
+                    self, 
```
Same thing, please fix indentation

skinan committed:
Resolve type-formatting issues and others

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
Merge branch 'integrate-clearml' of github.com:skinan/MONAI into integrate-clearml

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
Resolved reviews on pull-request

skinan committed:
Merge branch 'integrate-clearml' of github.com:skinan/MONAI into integrate-clearml

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

KumoLiu committed:
add pad transforms with unit tests for lazy resampling (#6031)

Part of #5991.
### Description

This PR adds the pad transforms from
https://github.com/Project-MONAI/MONAI/pull/5860 relate to lazy
resampling, and it belongs to the second part mentioned in
https://github.com/Project-MONAI/MONAI/issues/5991

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: KumoLiu <yunl@nvidia.com>

yiheng-wang-nv committed:
5991 Enable lazy resampling for SpatialResample (#6060)

A part of #5991 .

### Description

This PR adds the changes of `SpatialResample` from #5860 .

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: Yiheng Wang <vennw@nvidia.com>

skinan committed:
Fixed typehints based pull-request review.

rijobro committed:
include citation (#6063)

Add citation URL to README.

Signed-off-by: Richard Brown <33289025+rijobro@users.noreply.github.com>
Co-authored-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>

wyli committed:
update statshandler logging message (#6051)

### Description

The recommendation in `StatsHandler` of setting the global logging
config may not work in some cases (e.g. google colab and
https://github.com/Project-MONAI/MONAI/pull/5960)
https://docs.python.org/3/library/logging.html#logging.basicConfig


https://github.com/Project-MONAI/MONAI/blob/3eef61e82883e9bc6c08d26954fc86286df7d1b3/monai/handlers/stats_handler.py#L55

in general we shouldn't recommend changing the root logger directly.

This PR changes the message to tune the module level loggers, updated
example message will be:
```
.the effective log level of ignite.engine.engine.Engine higher than INFO, StatsHandler may not generate logs,
please use the following code before running the engine to enable it: 
                                                                                
import logging                                                              
logging.getLogger('ignite.engine.engine.Engine').setLevel(logging.INFO) 
```

the updated commands are tested in the colab example:

https://colab.research.google.com/drive/1boqy7ENpKrqaJoxFlbHIBnIODAs1Ih1T

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: Wenqi Li <wenqil@nvidia.com>

monai-bot committed:
auto updates (#6070)

Signed-off-by: monai-bot <monai.miccai2019@gmail.com>

Signed-off-by: monai-bot <monai.miccai2019@gmail.com>

johannesu committed:
Add affine_lps_to_ras argument to NrrdReader (#6074)

Fixes #6073

### Description

Add a affine_lps_to_ras argument to NrrdReader controlling whether to
convert the affine matrix from "LPS" to "RAS" or not. The naming and
functionality is identical to the argument used by PydicomReader and
ITKReader

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/`
folder.

Signed-off-by: Johannes Ulén <johannes.ulen@gmail.com>

Nic-Ma committed:
6028 6041 Upgrade ignite dependency to 0.4.11 (#6067)

Fixes #6028 .
Fixes #6041 .

### Description

This PR upgraded the ignite version of dependency to the latest 0.4.11.

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: Nic Ma <nma@nvidia.com>

yiheng-wang-nv committed:
6069 Add default metadata and logging values for bundle run (#6072)

Fixes #6069 .

### Description

This PR adds commonly used default values for `meta_file` and
`logging_file` of `monai.bundle.run` function.

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: Yiheng Wang <vennw@nvidia.com>

KumoLiu committed:
add crop transforms with unit tests for lazy resampling (#6068)

Part of #5991 .

### Description

This PR adds the crop transforms from
https://github.com/Project-MONAI/MONAI/pull/5860 relate to lazy
resampling, and it belongs to the second part mentioned in
https://github.com/Project-MONAI/MONAI/issues/5991

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: KumoLiu <yunl@nvidia.com>

wyli committed:
6066 pad mode (#6076)

Fixes #6066

### Description

prefer the pytorch backend as much as possible

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing
functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: Wenqi Li <wenqil@nvidia.com>

skinan committed:
Formatting, Unit Test added, type hinting fixed

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

skinan committed:
"output_uri" type hinting fixed

skinan committed:
Merge branch 'integrate-clearml' of github.com:skinan/MONAI into integrate-clearml

skinan committed:
type hinting fixed

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

skinan committed:
Type hinting

skinan committed:
Merge branch 'integrate-clearml' of github.com:skinan/MONAI into integrate-clearml

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

skinan committed:
Merge branch 'Project-MONAI:dev' into integrate-clearml

skinan committed:
review changes
